src/lib/auth.ts

```diff
@@ -40,14 +65,48 @@ export const NEXT_AUTH_OPTIONS: NextAuthOptions = {
   },
 };
 
-export async function getAuthenticatedAccount(): Promise<
-  Result<{ email: string; name: string }, "NO_SESSION">
-> {
+export async function isLoggedIn(accountType: AccountType): Promise<boolean> {
+  const session = await getLoggedInSession();
+  if (session === null) {
+    return false;
+  }
+  return accountType === session.accountType;
```

Comment:
This should attempt to load an actor of the session account type so that it is verified that the account exists and is not disabled or something.

Otherwise we might ding-dong between login page and (logged-in)/layout because the login page—which is shared by all account types—uses isLoggedIn, but the (logged-in)/layout layers check using getAuthenticated{type}Actor.